lib/ansible/plugins/action/ios.py

```diff
@@ -85,7 +85,7 @@ def run(self, tmp=None, task_vars=None):
         conn = Connection(socket_path)
         try:
             out = conn.get_prompt()
-            while to_text(out, errors='surrogate_then_replace').strip().endswith(')#'):
+            while to_text(out, errors='surrogate_then_replace').strip().endswith('config)#'):
```

Comment:
This won't work on any sublevel config... for example

```
an-ios-01(config)#interface GigabitEthernet0/2
an-ios-01(config-if)#
```

As written, this works because it will catch any level of config and `exit` out of it, but maybe we can instead check for something like `config.*)#` and `end` once to exit config?

Reply:
I totally missed the sub-level thing. Thanks! I'm updating the changes.